src/sound_openal.cpp

```diff
@@ -48,6 +48,25 @@ with this program; ifnot, write to the Free Software Foundation, Inc.,
 
 #define BUFFER_SIZE 30000
 
+std::shared_ptr<SoundManagerGlobal> g_sound_manager_global;
+
+typedef ::std::unique_ptr<ALCdevice, void (*)(ALCdevice *p)> unique_ptr_alcdevice;
+typedef ::std::unique_ptr<ALCcontext, void(*)(ALCcontext *p)> unique_ptr_alccontext;
```

Comment:
Is this notation required to support a specific compiler? `typedef std::unique_ptr<` would look better and is shorter.

Reply:
::type is for obscure cases with templates / ADL. No case where it would make a difference in Minetest though, stripping the :: .